src/prog_models/models/uav_model/uav_model.py

```diff
@@ -0,0 +1,567 @@
+        flightplan = trajectory.load.get_flightplan(fname=self.parameters['flight_file'])
+        lat, lon, alt, tstamps = flightplan['lat'], flightplan['lon'], flightplan['alt'], flightplan['timestamp']
+      elif self.parameters['flight_file'] != None and self.parameters['flight_plan'] != None:
+        raise ProgModelInputException("Too many flight plan inputs - please input either flight_plan dictionary or flight_file, not both.")
+      else:
+        raise ProgModelInputException("No flight plan information supplied. Please provide flight_plan or flight_file.")
+
+      aircraft1 = AircraftModels.build_model(name=self.parameters['aircraft_name'],
+                                              model=self.parameters['vehicle_model'],
+                                              payload=self.parameters['vehicle_payload'])
+      self.vehicle_model = aircraft1 
+
+      # Generate route
+      if len(tstamps) > 1:
+          # ETAs specified: 
+          # Check if speeds have been defined and warn user if so:
+          if self.parameters['cruise_speed'] is not None or self.parameters['ascent_speed'] is not None or self.parameters['descent_speed'] is not None:
+              warn("Speed values are ignored since ETAs were specified. To define speeds (cruise, ascent, descent) instead, do not specify ETAs.")
+          route_ = route.build(name=self.parameters['flight_name'], lat=lat, lon=lon, alt=alt, departure_time=tstamps[0],
+                                etas=tstamps,  # ETAs override any cruise/ascent/descent speed requirements. Do not feed etas if want to use desired speeds values.
+                                vehicle_max_speed = self.vehicle_model.dynamics['max_speed'],
+                                parameters = self.parameters)
+      else: 
+          # ETAs not specified:  
+          # Check that speeds have been provided:
+          if self.parameters['cruise_speed'] is None or self.parameters['ascent_speed'] is None or self.parameters['descent_speed'] is None or self.parameters['landing_speed'] is None:
+              raise ProgModelInputException("ETA or speeds must be provided. If ETAs are not defined, desired speed (cruise, ascent, descent, landing) must be provided.")  
+          route_ = route.build(name=self.parameters['flight_name'], lat=lat, lon=lon, alt=alt, departure_time=tstamps[0],
+                                parameters = self.parameters)
+
+      # Save final waypoint information for threshold_met and event_state 
+      coord_end = geom.Coord(route_.lat[0], route_.lon[0], route_.alt[0])
+      self.coord_transform = deepcopy(coord_end)
+      wypt_time_unix = [datetime.datetime.timestamp(route_.eta[iter]) - datetime.datetime.timestamp(route_.eta[0]) for iter in range(len(route_.eta))]
+      wypt_x, wypt_y, wypt_z = coord_end.geodetic2enu(route_.lat, route_.lon, route_.alt) # computing way-points in ENU in one shot
```

Comment:
here there was a for loop I substituted (functions in geom.Coord work with vectors).